src/importer/sources/retroarch_source.py

```diff
@@ -48,13 +49,15 @@ def generator_builder(self) -> SourceIterationResult:
                 continue
 
             for item in playlist_json["items"]:
-                # Select the core. Try the content's core first, then the playlist's
-                # default core.
-                core_path = item["core_path"]
-                if core_path == "DETECT":
-                    default_core = playlist_json["default_core_path"]
-                    if default_core:
-                        core_path = default_core
+                # Select the core.
+                # Try the content's core first, then the playlist's default core.
+                # If none can be used, warn the user and continue.
+                for core_path in (
+                    item["core_path"],
+                    playlist_json["default_core_path"],
+                ):
+                    if core_path != "DETECT":
+                        break
                     else:
```

Comment:
The `else` block should be de-indented, the goal is to inform the user if there was no break (i.e. no good core) and a neat python trick is the `for ... else` pattern :)

Reply:
I didn't know for-else existed! :O Thanks for the heads up!